### ISSUE TITLE
set id on search input element

### ISSUE THREAD

**Issue by dan-fig** [bithost-gmbh/ngx-mat-select-search#210]
**Describe the bug**
I cannot figure out how to add an id to the the mat-input controls that are created under the ngx-mat-select-search markup in the browser. I am able to successfully assign an id to the  ng-mat-select-search control itself. We have QA staff that would like to use this id on the mat-input for automation. Ideally, this id would remain consistently assigned, meaning if I go to the same page with the same controls, I can be assured the id values for each mat-input will be consistently set.

**To Reproduce**
Go to simple sample on stack: blitz;https://qgnirjmqa.github.stackblitz.io
1. Perform simple search by entering text 'H' in search box.
2. Using a browser debugger, view the markup in the browser. Note that the id values for the mat-input are autogenerated. I would like to assign the id values of the mat-input boxes.
![Screen Shot 2020-02-13 at 4 05 25 PM](https://user-images.githubusercontent.com/15680062/74482802-d719f280-4e7a-11ea-9253-76915f8e2ec1.png)

**Expected behavior**

 I would like to assign the id values of the mat-input boxes. Please provide a 'how-to' on best approach to do this.

**Screenshots**

**Desktop (please complete the following information):**
Chrome on Mac

**Additional context**
I am using Angular 5 and version 1.8 of the ngx-mat-select-search control. I'm stuck at this release of Angular for at least another year.
Thank you for your help!

**Comment by macjohnny** [comment]
@dan-fig the id is set by the `matInput` directive and is private api, see https://github.com/angular/components/blob/master/src/material/input/input.ts#L153-L160
however, you could set a custom `data-*` attribute using the `searchSelectInput: ElementRef` view child of the `<ngx-mat-select-search>` component whenever the `matSelect.openedChange` emits `true`, see this example:
```typescript
ngOnInit(){
    // set a custom data attribute to the ngx-mat-select-search input whenever the select is opened
    this.singleSelect.openedChange.pipe(
      filter(opened => opened),
      takeUntil(this._onDestroy)
    ).subscribe(() => {
      this.matSelectSearch.searchSelectInput.nativeElement.setAttribute('data-some-custom-e2e-attribute', '')
    })
}
```

try it here:
https://stackblitz.com/edit/github-lplndt-vslfsa?file=src/app/examples/01-single-selection-example/single-selection-example.component.ts 

![ngx-mat-select-search-custom-attribute](https://user-images.githubusercontent.com/5589029/74514163-54417800-4f0c-11ea-8c31-8a7eeb10fa3c.png)

**Comment by dan-fig** [comment]
This is a big help. Thank you

Sent from Yahoo Mail on Android 
 
  On Fri, Feb 14, 2020 at 2:29 AM, Esteban Gehring<notifications@github.com> wrote:   
Closed #210.
